prow/kube/prowjob.go

```diff
@@ -79,22 +79,74 @@ type ProwJob struct {
 }
 
 type ProwJobSpec struct {
-	Type    ProwJobType  `json:"type,omitempty"`
-	Agent   ProwJobAgent `json:"agent,omitempty"`
-	Cluster string       `json:"cluster,omitempty"`
-	Job     string       `json:"job,omitempty"`
-	Refs    *Refs        `json:"refs,omitempty"`
-
-	Report         bool   `json:"report,omitempty"`
-	Context        string `json:"context,omitempty"`
-	RerunCommand   string `json:"rerun_command,omitempty"`
-	MaxConcurrency int    `json:"max_concurrency,omitempty"`
-
+	// Type is the type of job and informs how
+	// the jobs is triggered
+	Type ProwJobType `json:"type,omitempty"`
+	// Agent determines which controller fulfills
+	// this specific ProwJobSpec and runs the job
+	Agent ProwJobAgent `json:"agent,omitempty"`
+	// Cluster is which Kubernetes cluster is used
+	// to run the job, only applicable for that
+	// specific agent
+	Cluster string `json:"cluster,omitempty"`
+	// Job is the name of the job
+	Job string `json:"job,omitempty"`
+	// Refs is the code under test, determined at
+	// runtime by Prow itself
+	Refs *Refs `json:"refs,omitempty"`
+	// ExtraRefs are auxiliary repositories that
+	// need to be cloned, determined from config
+	ExtraRefs []*Refs `json:"extra_refs,omitempty"`
+
+	// TimeoutMinutes is the number of minutes that
+	// the pod utilities will wait before aborting
+	// a job. Only applicable if decorating the PodSpec.
+	TimeoutMinutes int `json:"timeout_minutes,omitempty"`
```

Comment:
Did we agree on switching this to a duration or no?

Reply:
Want some direction on internal/external config for time.Duration since you can't deserialize into it